### PR TITLE
More readable implementation of tree_sort.py

### PR DESCRIPTION
**Pull request by Ashrockzzz2003**
## tree_sort_2

- makes use of the data structure Binary Tree to sort the list in O(nlogn) time.
- Binary Search Tree [BST]
  - For each node with value v - All values in the left subtree are < v - All values in the right subtree are > v
  - No Duplicate values.
- Logic:
  - Build a Binary Search Tree
  - Traverse the tree in increasing order.

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: sorts/tree_sort_2.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    - All values in the right subtree are > v
+  - No Duplicate values.
+- Logic:
+  - Build a Binary Search Tree
+  - Traverse the tree in increasing order.
+
+"""
+
+
+class Tree:
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:18]:
```suggestion
class TreeSort:
```

File: sorts/tree_sort_2.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        else:
+            self.left = None
+            self.right = None
+        return
+
+    # Empty nodes are None valued
+    def isempty(self) -> bool:
+        """
+        Returns true if tree is empty else False
+        >>> isempty([])
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:39]:
```suggestion
        >>> Tree().isempty()
```
It seems like `isempty` doesn't take any arguments, so why are you passing `[]`?

File: sorts/tree_sort_2.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  - Build a Binary Search Tree
+  - Traverse the tree in increasing order.
+
+"""
+
+
+class Tree:
+    # Binary Search Tree Data Structure
+    # Constructor
+    def __init__(self, initval: int | str | float = None) -> None:
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:21]: What exactly is `initval`? Could you be more descriptive please

File: sorts/tree_sort_2.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
+"""
+
+tree_sort_2
+
+- makes use of the data structure Binary Tree to sort the list in O(nlogn) time.
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:5]:
```suggestion
- Makes use of the data structure Binary Tree to sort the list in O(n log(n)) time.
```

File: sorts/tree_sort_2.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    # Empty nodes are None valued
+    def isempty(self) -> bool:
+        """
+        Returns true if tree is empty else False
+        >>> isempty([])
+        True
+        >>> isempty([1,2,3])
+        False
+        """
+        return self.value == None
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:44]:
```suggestion
        return self.value is None
```

File: sorts/tree_sort_2.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+    def isleaf(self) -> bool:
+        """
+        Returns true if leaf is a node
+        Suppose in [1,2,3,4,5,6,7,8,9,10]
+        4 had empty left and right branches.
+        >>> isleaf(4)
+        True
+        """
+        return self.value != None and self.left.isempty() and self.right.isempty()
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:54]:
```suggestion
        return self.value is not None and self.left.isempty() and self.right.isempty()
```

File: sorts/tree_sort_2.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+    # Inorder Traversal
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:56]: Put this comment within the function

File: sorts/tree_sort_2.py [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+  - No Duplicate values.
+- Logic:
+  - Build a Binary Search Tree
+  - Traverse the tree in increasing order.
+
+"""
+
+
+class Tree:
+    # Binary Search Tree Data Structure
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:19]: Put these comments within the class docstring

File: sorts/tree_sort_2.py [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+def tree_sort(list_data: list) -> list:
+    """
+    Returns a list array sorted in increasing order from the Binary Search Tree.
+    >>> tree_sort([23235,82107,35775,91961,4323,40556,76603,64302,27316,74372])
+    [4323, 23235, 27316, 35775, 40556, 64302, 74372, 76603, 82107, 91961]
+    >>> tree_sort([50,52,54,74,93,100,114,124,130,143])
+    [50, 52, 54, 74, 93, 100, 114, 124, 130, 143]
+    """
+    if len(list_data) == 0 or len(list_data) == 1:
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:120]:
```suggestion
    if len(list_data) <= 1:
```

File: sorts/tree_sort_2.py (hunk 4 above)
Comment (CaedenPH) [inline sorts/tree_sort_2.py:5]: Can you prove this?

File: sorts/tree_sort_2.py [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+        self.value = initval
+        if self.value:
+            self.left = Tree()
+            self.right = Tree()
+        else:
+            self.left = None
+            self.right = None
+        return
+
+    # Empty nodes are None valued
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:35]: Put this comment within a docstring in the function

File: sorts/tree_sort_2.py [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+            self.left = Tree()
+            self.right = Tree()
+
+        # Value already exists
+        if self.value == data:
+            return
+
+        if data < self.value:
+            self.left.insert(data)
+            return
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:106]:
```suggestion
```

File: sorts/tree_sort_2.py [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+        if data > self.value:
+            self.right.insert(data)
+            return
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:109]:
```suggestion
```
These `return` statements don't do anything

File: sorts/tree_sort_2.py [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+        >>> print(t)
+        [1,2,3]
+        """
+        return str(self.inorder())
+
+    # Insert new element
+    def insert(self, data: int | str | float) -> None:
+        """
+        Inserts the value data to the Binary Search Tree
+        Let
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:85]:
```suggestion

```

File: sorts/tree_sort_2.py [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+    Returns a list array sorted in increasing order from the Binary Search Tree.
+    >>> tree_sort([23235,82107,35775,91961,4323,40556,76603,64302,27316,74372])
+    [4323, 23235, 27316, 35775, 40556, 64302, 74372, 76603, 82107, 91961]
+    >>> tree_sort([50,52,54,74,93,100,114,124,130,143])
+    [50, 52, 54, 74, 93, 100, 114, 124, 130, 143]
+    """
+    if len(list_data) == 0 or len(list_data) == 1:
+        return list_data
+    tree = Tree()
+    for i in list_data:
```
Comment (CaedenPH) [inline sorts/tree_sort_2.py:123]:
```suggestion
    for data in list_data:
```

**Review by CaedenPH — CHANGES_REQUESTED** [review]
The doctests are failing. Please fix them and also prove that the algorithm sorts with `O(n log(n))` time

File: sorts/tree_sort_2.py (hunk 2 above)
Comment (Ashrockzzz2003) [reply sorts/tree_sort_2.py:39]: Sure. I'll update. Thank you!

File: sorts/tree_sort_2.py [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+
+
+class Tree:
+    # Binary Search Tree Data Structure
+    # Constructor
+    def __init__(self, initval: int | str | float = None) -> None:
+        """
+        An empty left and right branches will be created for every value inserted,
+        to perform better in recursive methods
+        """
```
[the remark this answers is not in the document]
Comment (Ashrockzzz2003) [reply sorts/tree_sort_2.py:25]: Sure. I'll update. Thank you!

File: sorts/tree_sort_2.py [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+    """
+    if len(list_data) == 0 or len(list_data) == 1:
+        return list_data
+    tree = Tree()
+    for i in list_data:
+        tree.insert(i)
+    return tree.inorder()
+
+
+if __name__ == "__main__":
```
[the remark this answers is not in the document]
Comment (Ashrockzzz2003) [reply sorts/tree_sort_2.py:128]: I'll check what's wrong! Thank you for that

File: sorts/tree_sort_2.py (hunk 3 above)
Comment (Ashrockzzz2003) [reply sorts/tree_sort_2.py:21]: `initval` is the value which the node of the tree holds

File: sorts/tree_sort_2.py (hunk 10 above)
Comment (Ashrockzzz2003) [reply sorts/tree_sort_2.py:35]: Sure, Thank you!

File: sorts/tree_sort_2.py (hunk 11 above)
Comment (Ashrockzzz2003) [reply sorts/tree_sort_2.py:106]: Why tho?

File: sorts/tree_sort_2.py (hunk 13 above)
Comment (Ashrockzzz2003) [reply sorts/tree_sort_2.py:85]: I can't understand what's this

File: sorts/tree_sort_2.py (hunk 14 above)
Comment (Ashrockzzz2003) [reply sorts/tree_sort_2.py:123]: Will do that!
Thank you

File: sorts/tree_sort_2.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (Ashrockzzz2003) [reply sorts/tree_sort_2.py:128]: - I've changed and started a new pull request that had 2 ticks. 
- Also, the doctests have been verified now and working. 
- This is the first time I'm contributing, What next should I do?

**Pull request closed without merging**